thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/common/thrift/logging/AbstractUnMaskingTProtocol.java

```diff
@@ -0,0 +1,547 @@
+        }
+        if (context instanceof OverwriteMetadataContext) {
+            return;
+        }
+        delegate.readListEnd();
+    }
+
+    @Override
+    public TSet readSetBegin() throws TException {
+        final MetadataContext context = stack.getFirst().resolve();
+        if (context instanceof OverwriteMetadataContext) {
+            final OverwriteMetadataContext overwriteContext = (OverwriteMetadataContext) context;
+            final FieldValueMetaData metaData = overwriteContext.valueMetaData();
+            final Object obj = overwriteContext.getObj();
+            if (metaData instanceof SetMetaData && obj instanceof Set) {
+                stack.push(new MultiOverrideMetadataContext((Set<?>) obj, (SetMetaData) metaData
+                ));
+                return new TSet(((SetMetaData) metaData).elemMetaData.type, ((Set<?>) obj).size());
+            }
+            // the context has been lost
+            stack.push(IgnoreOverwriteMetadataContext.INSTANCE);
+            return new TSet();
+        }
+        final TSet tSet = delegate.readSetBegin();
+        if (!(context instanceof CollectionMetadataContext)) {
+            stack.push(LostMetadataContext.INSTANCE);
+            return tSet;
+        }
+        final CollectionMetadataContext metadataContext = (CollectionMetadataContext) context;
+        stack.push(metadataContext);
+        return tSet;
+    }
+
+    @Override
+    public void readSetEnd() throws TException {
+        final MetadataContext context = stack.pop();
+        if (context instanceof MultiOverrideMetadataContext) {
+            checkState(((MultiOverrideMetadataContext) context).done(),
+                       "Collection <%s> is not fully consumed with stack: <%s>",
+                       context, stack);
+        }
+        if (context instanceof OverwriteMetadataContext) {
+            return;
+        }
+        delegate.readSetEnd();
+    }
+
+    @Override
+    public boolean readBool() throws TException {
+        final MetadataContext context = stack.getFirst().resolve();
+        if (context instanceof OverwriteMetadataContext) {
+            final OverwriteMetadataContext overwriteMetadataContext = (OverwriteMetadataContext) context;
+            return (Boolean) overwriteMetadataContext.getObj();
+        }
+        if (!(context instanceof PojoMetadataContext)) {
+            return delegate.readBool();
+        }
+        final PojoMetadataContext metadataContext = (PojoMetadataContext) context;
+        return tryUnmask(metadataContext, Boolean.class, delegate::readBool, delegate);
+    }
+
+    @Override
+    public byte readByte() throws TException {
+        final MetadataContext context = stack.getFirst().resolve();
+        if (context instanceof OverwriteMetadataContext) {
+            final OverwriteMetadataContext overwriteMetadataContext = (OverwriteMetadataContext) context;
+            return (Byte) overwriteMetadataContext.getObj();
```

Comment:
Question) Couldn't the value returned by `overwriteMetadataContext.getObj()` be cast to the primitive `(byte)`?